pkg/kudoctl/util/kudo/kudo.go

```diff
@@ -52,15 +53,15 @@ func NewClient(kubeConfigPath string, requestTimeout int64, validateInstall bool
 		return nil, clog.Errorf("could not get Kubernetes client: %s", err)
 	}
 
-	err = crd.NewInitializer().ValidateInstallation(kubeClient)
-	if err != nil {
-		// see above
-		if os.IsTimeout(err) {
-			return nil, err
-		}
+	result := verifier.NewResult()
+	err = crd.NewInitializer().VerifyInstallation(kubeClient, &result)
```

Comment:
Hmm, the old code doesn't do that, but shouldn't we run `VerifyInstallation` only if `validateInstall == true`? Closely related: If we run `VerifyInstallation` regardless of `validateInstall`, we should always return its error if there is on, similar to how it's done in the old code.

Reply:
Well - generally we *always* want to run VerifyInstallation. The `validateInstall` parameter is by default "true", and it allows a user to continue with the process even if the verification is returning errors. It's basically "I know what I'm doing, don't annoy me"-Button.

For returning an error - The new code behaves the same as the old one, as far as I can see: It aborts if the verification fails and the parameter is true, and continues if the parameter is false.

I have renamed the parameter though...

Reply:
The old code returned an error only for `os.IsTimeout`, regardless of `validateInstall`. IMO, we should do the same and always return an error if `VerifyInstallation` returns one, because it indicates something wrong with the cluster in general. For warnings/errors related to the CRDs specifically, `Result` is used and there `validateInstall` decides if an error is reported. WDYT?

Reply:
Oooh, that's what you mean. Right. Yeah, that makes sense.